macros/src/main/scala-2/org/typelevel/scalacoptions/literals.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.scalacoptions
+
+import org.typelevel.literally.Literally
+
+object literals {
+  implicit class ScalaVersionContext(val sc: StringContext) extends AnyVal {
+    def sver(args: Any*): ScalaVersion = macro ScalaVersionLiteral.make
+  }
+
+  object ScalaVersionLiteral extends Literally[ScalaVersion] {
+    def validate(c: Context)(s: String) = {
+      import c.universe._
+
+      if (ScalaVersion.fromString(s).isEmpty)
+        Left(s"incorrect or unsupported Scala version")
```

Comment:
Perhaps we should use the same error message String as `ScalaVersion.fromString.unsafe` here?